modal/sandbox.py

```diff
@@ -763,6 +764,15 @@ async def watch(
         async for event in _FileIO.watch(path, self._client, task_id, filter, recursive, timeout):
             yield event
 
+    @property
+    def tags(self) -> dict[str, str]:
+        """Tags (key-value pairs) associated with the Sandbox.
+
+        Tags can be used to filter results in `Sandbox.list`, and they are updated with the
+        `set_tags()` function.
+        """
+        return dict(self._tags)
```

Comment:
There is no default value for _tags, so you'd get an AttributeError when trying ot do this on a Sandbox that wasn't constructed via the `.list` method.

Unless we add the relevant metadata to the other hydration paths we'd have to keep some state and warn to the user why this field isn't available for those Sandboxes (adding another "color" of sandboxes) which is quite ugly